service/frontend/workflowHandler.go

```diff
@@ -2317,6 +2317,118 @@ func (wh *WorkflowHandler) ListOpenWorkflowExecutions(
 	return resp, nil
 }
 
+// ListArchivedWorkflowExecutions - retrieves archived info for closed workflow executions in a domain
+func (wh *WorkflowHandler) ListArchivedWorkflowExecutions(
+	ctx context.Context,
+	listRequest *gen.ListArchivedWorkflowExecutionsRequest,
```

Comment:
We talked about this supporting a simplified SQL syntax?

Reply:
I know. I will do that in another pr later as it will contain logic for parsing, translating, validating the SQL query (since it applies to all visibility archivers). I think for the first version of visibility archival it's ok not having SQL syntax. 

Or I can add a new API (or new field to the request struct) for the SQL syntax and change the visibility archiver interface correspondingly so that each archiver can enforce different rules on the SQL syntax. Thoughts?

Reply:
Updated frontend and visibility archiver interface to accept sql string. Created a query parser interface for filestore. Will implement this interface later.